### PR TITLE
Update console related documentation

### LINKED ISSUE #1688: Minor improvements in Getting Started
#### Summary

Minor improvements in Getting Started 

#### Why do we need this ?

To smoothen the first steps with The Things Stack

#### What is already there? What do you see now?

- Manual linking of AS to NS with creation of API key
- No separate guide for creating a device

#### What is missing? What do you want to see?

- Change Getting Started to use the automatic linking feature (enabled by default); remove obsolete steps
- Create a separate step or page for creating devices, to which we can link directly. This should also explain the MAC and PHY versions

#### How do you propose to document this?

See above

#### Can you do this yourself and submit a Pull Request?

Can review

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

Closes #1688 
Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/783

#### Changes

- Extract each step of the Console getting started guide into a separate page.
- Added a note about PHY vs MAC versions.

#### Notes for Reviewers

None.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: doc/content/guides/getting-started/console/create-gateway/_index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+{{< figure src="gateway-creation.png" alt="Gateway creation" >}}
+
+Your gateway will be created and you will be redirected to the gateway overview page of your newly created gateway.
+
+{{< figure src="gateway-overview.png" alt="Gateway overview" >}}
+
+You can now connect your gateway to {{% tts %}}.
+
+### Create Gateway API Key
```
Comment (neoaggelos) [inline doc/content/guides/getting-started/console/create-gateway/_index.md]: Hijacking to mention: There are a few gateway connection guides that link to this part of the initial doc (e.g. https://github.com/TheThingsNetwork/lorawan-stack/blob/master/doc/content/guides/connecting-gateways/nasysoutdoorgateway/_index.md#registration)

The links in those files should be updated as well. There may be more references else-where

**Comment by kschiffer** [comment]
Blocking #2056

File: doc/content/guides/getting-started/console/create-gateway/_index.md (hunk 1 above)
Comment (kschiffer) [reply doc/content/guides/getting-started/console/create-gateway/_index.md]: Good catch, thanks!

**Comment by kschiffer** [comment]
- Added section about application linking
- Fixed getting started guides references
- Updated screenshots
- Added console config reference

File: doc/content/reference/configuration/console.md [hunk 2]
```diff
@@ -3,3 +3,65 @@ title: "Console Options"
...
+- `console.ui.is.enabled`: Enable this API
+- `console.ui.js.base-url`: Base URL to the HTTP API
+- `console.ui.js.enabled`: Enable this API
+- `console.ui.ns.base-url`: Base URL to the HTTP API
+- `console.ui.ns.enabled`: Enable this API
+- `console.ui.qrg.base-url`: Base URL to the HTTP API
+- `console.ui.qrg.enabled`: Enable this API
+
+
+### File includes
```
Comment (johanstokking) [inline doc/content/reference/configuration/console.md]: Title case

File: doc/content/reference/configuration/console.md [hunk 3]
```diff
@@ -3,3 +3,65 @@ title: "Console Options"
...
+- `console.ui.is.base-url`: Base URL to the HTTP API
+- `console.ui.is.enabled`: Enable this API
+- `console.ui.js.base-url`: Base URL to the HTTP API
+- `console.ui.js.enabled`: Enable this API
+- `console.ui.ns.base-url`: Base URL to the HTTP API
+- `console.ui.ns.enabled`: Enable this API
+- `console.ui.qrg.base-url`: Base URL to the HTTP API
+- `console.ui.qrg.enabled`: Enable this API
+
+
```
Comment (johanstokking) [inline doc/content/reference/configuration/console.md]: Remove duplicate empty line

File: doc/content/reference/configuration/console.md [hunk 4]
```diff
@@ -3,3 +3,65 @@ title: "Console Options"
...
+- `console.oauth.client-secret`: The OAuth client secret
+- `console.oauth.token-url`: The OAuth Token Exchange URL
+
+## Frontend Setup
+
+You can change various values that will be passed to or used by the frontend JacaScript logic:
+
+### Component Configuration
+
+The Console needs to know how the individual {{% tts %}} components are configured, so it can execute API calls correctly. Likewise, **disabled components will cause respective options and menus in the Console frontend to be disabled**.
```
Comment (johanstokking) [inline doc/content/reference/configuration/console.md]: This sounds like a warning while it's just a sensible feature to me

File: doc/content/reference/configuration/console.md [hunk 5]
```diff
@@ -3,3 +3,65 @@ title: "Console Options"
...
+The Console app uses the [OAuth 2.0 authorization flow](https://en.wikipedia.org/wiki/OAuth#OAuth_2.0_2) to authorize actions in the backend. You can customize the authorization parameters if necessary:
+
+- `console.oauth.authorize-url`: The OAuth Authorize URL
+- `console.oauth.client-id`: The OAuth client ID
+- `console.oauth.client-secret`: The OAuth client secret
+- `console.oauth.token-url`: The OAuth Token Exchange URL
+
+## Frontend Setup
+
+You can change various values that will be passed to or used by the frontend JacaScript logic:
```
Comment (johanstokking) [inline doc/content/reference/configuration/console.md]: typo

File: doc/content/reference/configuration/console.md [hunk 6]
```diff
@@ -3,3 +3,65 @@ title: "Console Options"
 description: ""
 weight: 7
 ---
+
+## Console mount
+
+The Console can be served under any arbitrary path on your server
+
+- `console.mount string`: Path on the server where the Console will be served
```
Comment (johanstokking) [inline doc/content/reference/configuration/console.md]: Remove `string`

File: doc/content/reference/configuration/console.md [hunk 7]
```diff
@@ -3,3 +3,65 @@ title: "Console Options"
...
+## Frontend Setup
+
+You can change various values that will be passed to or used by the frontend JacaScript logic:
+
+### Component Configuration
+
+The Console needs to know how the individual {{% tts %}} components are configured, so it can execute API calls correctly. Likewise, **disabled components will cause respective options and menus in the Console frontend to be disabled**.
+
+- `console.ui.as.base-url`: Base URL to the HTTP API
+- `console.ui.as.enabled`: Enable this API
```
Comment (johanstokking) [inline doc/content/reference/configuration/console.md]: Is it the API that's enabled or the UI functionality?

File: doc/content/reference/configuration/console.md [hunk 8]
```diff
@@ -3,3 +3,65 @@ title: "Console Options"
...
+- `console.ui.ns.enabled`: Enable this API
+- `console.ui.qrg.base-url`: Base URL to the HTTP API
+- `console.ui.qrg.enabled`: Enable this API
+
+
+### File includes
+
+You can control which CSS and JS files are included into the rendered HTML of the Console. Note that these files must be present in the `/public` folder during runtime. 
+
+- `console.ui.js-file`: The names of the JS file
```
Comment (johanstokking) [inline doc/content/reference/configuration/console.md]: JS vs JavaScript in this doc

File: doc/content/reference/configuration/console.md [hunk 9]
```diff
@@ -3,3 +3,65 @@ title: "Console Options"
...
+- `console.ui.css-file`: The names of the CSS files
+
+### Assets and custom branding
```
Comment (johanstokking) [inline doc/content/reference/configuration/console.md]: Title case

File: doc/content/reference/configuration/console.md [hunk 10]
```diff
@@ -3,3 +3,64 @@ title: "Console Options"
 description: ""
 weight: 7
 ---
+
+## Console mount
```
Comment (johanstokking) [inline doc/content/reference/configuration/console.md]: Mount

**Pull request merged**